stacks/spica/src/passport/pages/api-key-add/api-key-add.component.ts

```diff
@@ -0,0 +1,47 @@
+import {Component, OnInit, TemplateRef, ViewChild} from "@angular/core";
+import {ApiKey, emptyApiKey} from "src/passport/interfaces/api-key";
+import {Router, ActivatedRoute} from "@angular/router";
+import {filter, switchMap, take, tap} from "rxjs/operators";
+import {MockService} from "src/passport/services/api-key.service";
+
+@Component({
+  selector: "app-api-key-add",
+  templateUrl: "./api-key-add.component.html",
+  styleUrls: ["./api-key-add.component.scss"]
+})
+export class ApiKeyAddComponent implements OnInit {
+  @ViewChild("toolbar", {static: true}) toolbar: TemplateRef<any>;
+  public apiKey: ApiKey = emptyApiKey();
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private apiKeyService: MockService
+  ) {}
+
+  ngOnInit() {
+    this.getApiKeyFromId();
+  }
+
+  getApiKeyFromId() {
+    this.activatedRoute.params
+      .pipe(
+        filter(params => params.id),
+        switchMap(params => this.apiKeyService.get(params.id)),
+        take(1)
+      )
+      .subscribe(apiKey => {
+        this.apiKey = apiKey;
+      });
+  }
```

Comment:
This logic will fail silently when the id parameter changes reactively.

You should move this logic into the ngOnInit hook and handle reactively by replacing the take(1) pipe with takeUntil pipe and unsubscribe when the view destroyed.

Reply:
Can you give me an example of this **when the id parameter changes reactively** scenario ?